client/src/main/scala/com/dwolla/cloudflare/domain/model/ZoneSettings.scala

```diff
@@ -6,38 +6,54 @@ import io.circe.generic.semiauto._
 
 object ZoneSettings {
   sealed trait CloudflareTlsLevel
+  object CloudflareTlsLevel {
+    case object Off extends CloudflareTlsLevel
+    case object FlexibleTls extends CloudflareTlsLevel
+    case object FullTls extends CloudflareTlsLevel
+    case object FullTlsStrict extends CloudflareTlsLevel
+    case object StrictTlsOnlyOriginPull extends CloudflareTlsLevel
 
-  case object Off extends CloudflareTlsLevel
-  case object FlexibleTls extends CloudflareTlsLevel
-  case object FullTls extends CloudflareTlsLevel
-  case object FullTlsStrict extends CloudflareTlsLevel
-  case object StrictTlsOnlyOriginPull extends CloudflareTlsLevel
-
-  implicit val cloudflareTlsLevelEncoder: Encoder[CloudflareTlsLevel] =
-    encoderBuilder[CloudflareTlsLevel] {
-      case Off => "off"
-      case FlexibleTls => "flexible"
-      case FullTls => "full"
-      case FullTlsStrict => "strict"
-      case StrictTlsOnlyOriginPull => "origin_pull"
-    }
+    implicit val cloudflareTlsLevelEncoder: Encoder[CloudflareTlsLevel] =
+      encoderBuilder[CloudflareTlsLevel] {
+        case Off => "off"
+        case FlexibleTls => "flexible"
+        case FullTls => "full"
+        case FullTlsStrict => "strict"
+        case StrictTlsOnlyOriginPull => "origin_pull"
+      }
+  }
 
   sealed trait CloudflareSecurityLevel
+  object CloudflareSecurityLevel {
+    case object EssentiallyOff extends CloudflareSecurityLevel
+    case object Low extends CloudflareSecurityLevel
+    case object Medium extends CloudflareSecurityLevel
+    case object High extends CloudflareSecurityLevel
+    case object UnderAttack extends CloudflareSecurityLevel
+
+    implicit val cloudflareSecurityLevelEncoder: Encoder[CloudflareSecurityLevel] =
+      encoderBuilder[CloudflareSecurityLevel] {
+        case EssentiallyOff => "essentially_off"
+        case Low => "low"
+        case Medium => "medium"
+        case High => "high"
+        case UnderAttack => "under_attack"
+      }
+  }
+
+  sealed trait CloudflareWaf
+  object CloudflareWaf {
+    case object Off extends CloudflareWaf
```

Comment:
There was some overlap in the types of settings available for `CloudflareWaf` and other related traits. For example,  `Off` is a valid value for both `CloudflareTlsLevel` and `CloudflareWaf`. To work around this, I enclosed case class definitions in their own singleton objects.